test/session-serialization.test.ts

```diff
@@ -127,15 +127,17 @@ describe('Authenticator session serialization', () => {
     expect(homeResponse.body).toContain('Failed to deserialize user out of session')
 
     // can't serve other requests either because the secure session decode fails, which would populate request.user even for unauthenticated requests
-    const otherResponse = await server.inject({
-      url: '/unprotected',
-      headers: {
-        cookie: loginResponse.headers['set-cookie'],
-      },
-      method: 'GET',
-    })
-
-    expect(otherResponse.statusCode).toEqual(500)
+    try {
+      await server.inject({
+        url: '/unprotected',
+        headers: {
+          cookie: homeResponse.headers['set-cookie'],
+        },
+        method: 'GET',
+      })
+    } catch (error) {
+      expect(error.message).toContain('Failed to deserialize user out of session')
```

Comment:
Jest has an `await expect(somePromise).rejects.toContain` helper if you want to make this a bit less verbose, the jest eslint rules recommend this form